src/Microsoft.AspNet.SignalR.JS/jquery.signalR.core.js

```diff
@@ -689,69 +695,112 @@
                         return;
                     }
 
-                    keepAliveData = connection._.keepAliveData;
-                    connection.appRelativeUrl = res.Url;
-                    connection.id = res.ConnectionId;
-                    connection.token = res.ConnectionToken;
-                    connection.webSocketServerUrl = res.WebSocketServerUrl;
+                    // redirect case
+                    if (res.AccessToken && res.Url) {
+                        // initiate the actual connection to res.Url with res.AccessToken
+                        connection.accessToken = res.AccessToken;
+                        setConnectionUrl(connection, res.Url);
```

Comment:
This pattern will only support one cycle of redirection, whereas ASP.NET Core SignalR supports up to 100 levels of redirection. However, that's because we tried to view the ASP.NET Core feature as "redirection" rather than just "SignalR Service Support". For ASP.NET SignalR, I'm much less worried about the general-purpose redirection feature, so I think this is OK. Thoughts @davidfowl ?

Reply:
Also, can we log here?

```
connection.log("Received redirect to: " + res.Url);
```

Reply:
Practically there's only ever one but it would be nice to be consistent.